# Candy Conundrum 

Candy Conundrum
I have two bowls of candy. One is supposed to be filled with candy-covered chocolates, the other filled with fruit-flavored candies with candy shells. Unfortunately, someone has nefariously mixed up the contents of the two bowls, mixing a portion of one bowl into the second, and vice versa. Even worse, the candies are visually indistinguishable, and can only be sorted out by taste.

I know that one of the bowls (A) has candies in the following proportions: 
A ~ {2/3 chocolate, 1/3 fruity}, 
B ~ {1/4 chocolate, 3/4 fruity}. 
Assume for the questions below that both bowls are large enough that small samples will not have a practical effect on the distribution of candies.

Input your answers as fractions or as decimals with at least four digits of precision.

I select a bowl at random and choose a random candy. It's a chocolate candy! What is the probability that I've picked up bowl A (the one weighted towards chocolates)?

In [34]:
AChoc = 2/3
AFruit = 1/3
BChoc = 1/4
BFruit = 3/4

P(c)      = ~.25 // assumed the unconditional probabilities to be 1/2 for each bowl

P(A)      = .5 

P(C|A)    = .6667 // decimal equivalent of 2/3 

P(c)      = (.5 * BChoc) + (.5 * AChoc)

In [32]:
Pc = (.5 * BChoc) + (.5 * AChoc)
print(Pc)

0.4583333333333333


P(A|c)    = P(C|A) * P(A) / P(c)

P(A|c)    = .6667  *.5 / .4583333333333333

P(A|c)    = 0.33335   / .4583333333333333

P(A|c)    = 0.33335  / .4583333333333333
    
P(A|c)    = 8/11 // .7272

A ~ {2/3 chocolate, 1/3 fruity}, = 2/3

B ~ {1/4 chocolate, 3/4 fruity}  =/4

Continuing with the same bowl as before, I select two more candies. They are both fruity candies. What is the probability that I've picked up bowl A (the one weighted towards chocolates) now? (To clarify, there are three candies drawn in all: one chocolate and two fruity.)

P(A|cff) = P(cff|A) * P(A) / P(cff)

In [18]:
Acff = AChoc * AFruit * AFruit
Bcff = BChoc * BFruit * BFruit
print(Acff)
print(Bcff)

0.07407407407407407
0.140625


P(A|cff) = (ACFF * .5) / Pcff

P(cff|A) = P(c|A) * p(f|A) * p(f|A) 

In [20]:
Pcff = (ACFF *.5) + (BCFF *.5)
print(Pcff)

0.10734953703703703


The denominator `P(cff)` is the probability of pulling cff from a single bowl (either A or B). 
So it is the probability of pulling cff from A times the probability of choosing A  
*plus* the probability of pulling cff from B times the probability of choosing B.

In [21]:
PAcff = (ACFF * .5) / Pcff
print(PAcff)

0.34501347708894875
